[BUG] Integration tests failing on Spark 3.2 due to BinaryType

After #6298 integration tests on Spark 3.2+ have been failing due to plans not being on the GPU due to BinaryType.  Some expression overrides in the Spark 3.2+ shims were missed.  Sample failures:
```
05:35:40  FAILED ../../src/main/python/conditionals_test.py::test_case_when[Binary] - p...
05:35:40  FAILED ../../src/main/python/conditionals_test.py::test_case_when[Array(Binary)]
05:35:40  FAILED ../../src/main/python/array_test.py::test_array_item[Array(Binary)] - ...
05:35:40  FAILED ../../src/main/python/array_test.py::test_make_array[Binary] - pyspark...
05:35:40  FAILED ../../src/main/python/array_test.py::test_array_element_at[Array(Binary)]
05:35:40  FAILED ../../src/main/python/conditionals_test.py::test_coalesce[Binary] - py...
05:35:40  FAILED ../../src/main/python/conditionals_test.py::test_if_else[Binary] - pys...
05:35:40  FAILED ../../src/main/python/generate_expr_test.py::test_explode_litarray[Binary][IGNORE_ORDER({'local': True})]
05:35:40  FAILED ../../src/main/python/generate_expr_test.py::test_posexplode_litarray[Binary][IGNORE_ORDER({'local': True})]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_date_keys[Map(Date(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_timestamp_keys[Map(Timestamp(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_string_keys[Map(String(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Byte(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Short(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Integer(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Long(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Float(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Double(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_numeric_keys[Map(Decimal(not_null)(18,0),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_date_keys[Map(Date(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_get_map_value_timestamp_keys[Map(Timestamp(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Byte(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Short(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Integer(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Long(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Float(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Double(not_null),Binary)]
05:35:40  FAILED ../../src/main/python/map_test.py::test_element_at_map_numeric_keys[Map(Decimal(not_null)(18,0),Binary)]
```

--- a/docs/supported_ops.md
+++ b/docs/supported_ops.md
@@ -530,11 +530,11 @@ Accelerator supports are described below.
 <td>S</td>
 <td>S</td>
 <td>S</td>
+<td>S</td>
 <td><b>NS</b></td>
-<td><b>NS</b></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, UDT</em></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, UDT</em></td>
-<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, UDT</em></td>
+<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types CALENDAR, UDT</em></td>
+<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types CALENDAR, UDT</em></td>
+<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types CALENDAR, UDT</em></td>
 <td><b>NS</b></td>
 </tr>
 <tr>

--- a/sql-plugin/src/main/311until320-nondb/scala/com/nvidia/spark/rapids/shims/Spark31XShims.scala
+++ b/sql-plugin/src/main/311until320-nondb/scala/com/nvidia/spark/rapids/shims/Spark31XShims.scala
@@ -111,7 +111,7 @@ abstract class Spark31XShims extends SparkShims with Spark31Xuntil33XShims with 
     GpuOverrides.exec[CustomShuffleReaderExec](
       "A wrapper of shuffle query stage",
       ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.ARRAY +
-          TypeSig.STRUCT + TypeSig.MAP).nested(), TypeSig.all),
+          TypeSig.STRUCT + TypeSig.MAP + TypeSig.BINARY).nested(), TypeSig.all),
       (exec, conf, p, r) => new GpuCustomShuffleReaderMeta(exec, conf, p, r))
 
   override def findOperators(plan: SparkPlan, predicate: SparkPlan => Boolean): Seq[SparkPlan] = {

--- a/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala
+++ b/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/Spark320PlusShims.scala
@@ -60,7 +60,7 @@ trait Spark320PlusShims extends SparkShims with RebaseShims with Logging {
   override final def aqeShuffleReaderExec: ExecRule[_ <: SparkPlan] = exec[AQEShuffleReadExec](
     "A wrapper of shuffle query stage",
     ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.ARRAY +
-      TypeSig.STRUCT + TypeSig.MAP).nested(), TypeSig.all),
+      TypeSig.STRUCT + TypeSig.MAP + TypeSig.BINARY).nested(), TypeSig.all),
     (exec, conf, p, r) => new GpuCustomShuffleReaderMeta(exec, conf, p, r))
 
   override final def sessionFromPlan(plan: SparkPlan): SparkSession = {
@@ -191,8 +191,9 @@ trait Spark320PlusShims extends SparkShims with RebaseShims with Logging {
       ExprChecks.projectAndAst(
         TypeSig.astTypes,
         (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.CALENDAR
-            + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT + TypeSig.ansiIntervals)
-            .nested(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 +
+            + TypeSig.BINARY + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT
+            + TypeSig.ansiIntervals)
+            .nested(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.BINARY +
                 TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT),
         TypeSig.all),
       (lit, conf, p, r) => new LiteralExprMeta(lit, conf, p, r)),

--- a/tools/src/main/resources/supportedExecs.csv
+++ b/tools/src/main/resources/supportedExecs.csv
@@ -14,7 +14,7 @@ SortExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,S,NS,PS,PS,PS,NS
 SubqueryBroadcastExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,S,S,PS,PS,PS,S
 TakeOrderedAndProjectExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,NS,NS,PS,PS,PS,NS
 UnionExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,NS,NS,PS,PS,PS,NS
-CustomShuffleReaderExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,NS,NS,PS,PS,PS,NS
+CustomShuffleReaderExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,S,NS,PS,PS,PS,NS
 HashAggregateExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,NS,NS,PS,PS,PS,NS
 ObjectHashAggregateExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,PS,NS,PS,PS,PS,NS
 SortAggregateExec,S,None,Input/Output,S,S,S,S,S,S,S,S,PS,S,S,S,PS,NS,PS,PS,PS,NS

